baseapp/baseapp_test.go

```diff
@@ -444,95 +444,62 @@ func TestVersionSetterGetter(t *testing.T) {
 	require.Equal(t, versionString, string(res.Value))
 }
 
-func TestLoadVersionInvalid(t *testing.T) {
-	logger := log.NewNopLogger()
-	pruningOpt := baseapp.SetPruning(storetypes.PruneNothing)
-	db := dbm.NewMemDB()
-	name := t.Name()
-	app := baseapp.NewBaseApp(name, logger, db, pruningOpt)
-
-	err := app.LoadLatestVersion()
-	require.Nil(t, err)
-
-	// require error when loading an invalid version
-	err = app.LoadVersion(-1)
-	require.Error(t, err)
-
-	header := tmproto.Header{Height: 1}
-	app.BeginBlock(abci.RequestBeginBlock{Header: header})
-	res := app.Commit()
-	commitID1 := storetypes.CommitID{Version: 1, Hash: res.Data}
-
-	// create a new app with the stores mounted under the same cap key
-	app = baseapp.NewBaseApp(name, logger, db, pruningOpt)
-
-	// require we can load the latest version
-	err = app.LoadVersion(1)
-	require.Nil(t, err)
-	testLoadVersionHelper(t, app, int64(1), commitID1)
-
-	// require error when loading an invalid version
-	err = app.LoadVersion(2)
-	require.Error(t, err)
-}
-
 func TestLoadVersionPruning(t *testing.T) {
 	logger := log.NewNopLogger()
-	pruningOptions := storetypes.PruningOptions{
+	pruningOptions := stypes.PruningOptions{
 		KeepRecent: 2,
 		KeepEvery:  3,
 		Interval:   1,
 	}
 	pruningOpt := baseapp.SetPruning(pruningOptions)
-	db := dbm.NewMemDB()
+	schemaOpt := baseapp.SetSubstores(sdk.NewKVStoreKey("key1"))
+	db := memdb.NewDB()
 	name := t.Name()
-	app := baseapp.NewBaseApp(name, logger, db, pruningOpt)
-
-	// make a cap key and mount the store
-	capKey := sdk.NewKVStoreKey("key1")
-	app.MountStores(capKey)
+	app := baseapp.NewBaseApp(name, logger, db, pruningOpt, schemaOpt)
 
-	err := app.LoadLatestVersion() // needed to make stores non-nil
+	err := app.Init() // needed to make stores non-nil
 	require.Nil(t, err)
 
-	emptyCommitID := storetypes.CommitID{}
+	emptyCommitID := stypes.CommitID{}
 
 	// fresh store has zero/empty last commit
 	lastHeight := app.LastBlockHeight()
 	lastID := app.LastCommitID()
 	require.Equal(t, int64(0), lastHeight)
 	require.Equal(t, emptyCommitID, lastID)
 
-	var lastCommitID storetypes.CommitID
+	var lastCommitID stypes.CommitID
 
 	// Commit seven blocks, of which 7 (latest) is kept in addition to 6, 5
 	// (keep recent) and 3 (keep every).
 	for i := int64(1); i <= 7; i++ {
 		app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: i}})
 		res := app.Commit()
-		lastCommitID = storetypes.CommitID{Version: i, Hash: res.Data}
+		lastCommitID = stypes.CommitID{Version: i, Hash: res.Data}
 	}
 
+	// TODO: behavior change -
+	// CacheMultiStoreWithVersion returned no error on missing version (?)
```

Comment:
I will collect these changes in behavior, to document them and review possible breakages